docs/developers-guide.md

```diff
@@ -3,319 +3,131 @@
 for a public OpenStack Cloud through the [OpenStack Passport](https://www.openstack.org/passport)
 program, or you can install a small private development cloud with
 [DevStack](https://docs.openstack.org/devstack/latest/).
 
-Once you have obtained access to an OpenStack cloud, you will need to start a development
-VM. The rest of this guide assumes a CentOS 7 cloud image, but should be easily transferrable
-to whatever development environment you prefer. You will need to have your cloud credentials
-loaded into your environment. For example, I use this `openrc` file:
-
-```
-export OS_PROJECT_DOMAIN_NAME=Default
-export OS_USER_DOMAIN_NAME=Default
-export OS_DOMAIN_ID=<domain_id_that_matches_name>
-export OS_PROJECT_NAME=<project_name>
-export OS_TENANT_NAME=<project_name>
-export OS_TENANT_ID=<project_id_that_matches_name>
-export OS_USERNAME=<username>
-export OS_PASSWORD=<password>
-export OS_AUTH_URL=http://<openstack_keystone_endpoint>/v3
-export OS_INTERFACE=public
-export OS_IDENTITY_API_VERSION=3
-export OS_REGION_NAME=<region_name>
-```
-
-The specific values you use will vary based on your particular environment. You may
-notice that several values are aliases of one another. This is in part because the
-values expected by the OpenStack client and
-[Gopher Cloud](https://github.com/gophercloud/gophercloud) are slightly different,
-especially with respect to the change from using `tenant` to `project`. One of our
-development goals is to make this setup easier and more consistent.
+openstack-cloud-controller-manager relies on [Octavia](https://docs.openstack.org/octavia/latest/) to create Kubernetes Service of type LoadBalancer, so make sure Octavia is available in the OpenStack cloud. For public cloud, you need to check the cloud documentation. For devstack, refer to Octavia deployment [quick start guide](https://docs.openstack.org/octavia/latest/contributor/guides/dev-quick-start.html).
 
-### Docker
+DevStack is recommended for cloud provider openstack development as some features are dependent on latest versions of openstack services, which is not always the case for public cloud providers. To get better performance, it's recommended to enable [nested virtualization](https://docs.openstack.org/devstack/latest/guides/devstack-with-nested-kvm.html) on the devstack host.
 
-Your cloud instance will need to have Docker installed. To install, please refer to [Docker Container runtime](https://kubernetes.io/docs/setup/production-environment/container-runtimes/#docker).
+### Kubernetes cluster
+Deploy a kubernetes cluster in the openstack cloud.
 
-### Development tools
+There are many deployment tools out there to set up a kubernetes cluster such as kubeadm, kubespray, kops, etc. choose the one you are familiar with and easy to customize. Make sure the cluster is configured with [external cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller).
```

Comment:
I knew it's overkill, as ecosystem buildup, can you link 
https://github.com/kubernetes-sigs/cluster-api-provider-openstack as well?

Reply:
done